Mathlib/RepresentationTheory/Tannaka.lean

```diff
@@ -147,6 +147,56 @@ def algHomOfRightFDRepComp (η : Aut (forget k G)) : (G → k) →ₐ[k] (G →
   apply_fun (fun x ↦ (x.hom.app rightFDRep).hom (1 : G → k)) at this
   exact this
 
+variable [DecidableEq G]
+
+/-- Auxiliary map for the proof of `toRightFDRepComp_inj`. -/
+@[simps]
+def auxLinearMap {X : FDRep k G} (v : X) : (G → k) →ₗ[k] X where
+  toFun f := ∑ s : G, (f s) • (X.ρ s⁻¹ v)
+  map_add' _ _ := by
+    simp only [add_apply, add_smul]
+    exact sum_add_distrib
+  map_smul' _ _ := by
+    simp only [smul_apply, smul_eq_mul, RingHom.id_apply, smul_sum, smul_smul]
+
+@[simp]
+lemma auxLinearMap_single_id {X : FDRep k G} (v : X) :
```

Comment:
```suggestion
lemma _root_.FDRep.sumSMulInv_single_id [DecidableEq G] {X : FDRep k G} (v : X) :
```
Rename, and add the decidability hypothesis when it's need in the theorem *statement*. To be honest you might be better off closing the TannakaDuality namespace completely and moving into the FDRep namespace for these?